### ISSUE TITLE
[Java] Execute PMD Custom RuleSet using PMD command line

### ISSUE THREAD

**Issue by Neha-Dhonde** [pmd/pmd#589]
Please, prefix the report title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*

**Rule Set:**

**Description:**
Need help on executing the PMD custom Ruleset on java source code directory via command line. I am able to execute the ruleset on a single file through the Designer. But not able to run the PMD ruleset on entire file source code using CLI.
**Code Sample demonstrating the issue:**

```

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
Hi @Neha-Dhonde ,
this should work with the following command line, given that your ruleset file is called "custom-rules.xml":

    ./bin/run.sh pmd -R custom-rules.xml -d src-folder/ -f text

On the [pmd documation page](https://pmd.github.io/pmd-5.8.1/usage/running.html) you can see a more exhaustive documentation about the various flags and possible output formats.

But maybe you need help in putting your xpath expression developed with the Designer into a ruleset file? This part is indeed missing in the [XPath Rule documentatio](https://pmd.github.io/pmd-5.8.1/customizing/xpathruletutorial.html).
Your ruleset "custom-rules.xml" could look like this:

```xml
<?xml version="1.0"?>

<ruleset name="Custom Rules"
    xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
  <description>
Custom rules
  </description>

    <rule name="My Rule"
          language="java"
          message="violation message"
          class="net.sourceforge.pmd.lang.rule.XPathRule">
     <description>
Rule Description
     </description>
     <priority>3</priority>
     <properties>
         <property name="xpath">
             <value><![CDATA[
--- here comes your XPath expression
]]></value>
         </property>
     </properties>
     <example>
 <![CDATA[
public class ExampleCode {
    public void foo() {
    }
}
]]>
     </example>
</rule>
</ruleset>
```

Regards,
Andreas

**Comment by Neha-Dhonde** [comment]
Thanks Andreas! Your inputs were of great help. I am able to run PMD through command line now. 
The issue was, i was using an older version (pmd-bin-4.2.6) of PMD which seemed to be not working as expected. I then downloaded the latest version pmd-bin-5.8.1, which is now working as desired.

**Comment by jsotuyod** [comment]
@Neha-Dhonde glad you managed to work it out.
